public/editor-client/src/api/index.ts

```diff
@@ -1206,3 +1217,175 @@ export const updateGlobalBlocks = async (
+
+    return {
+      blocks: res.collections,
+      categories: res.categories,
+      styles: res.styles
+    };
+  }
+
+  throw new Error(t("Failed to load kits"));
+};
+
+export const getKitData = async (
+  url: string,
+  kitId: string,
+  id: string
+): Promise<DefaultBlock> => {
+  const fullUrl = makeUrl(url, {
+    project_id: kitId,
+    page_slug: id
+  });
+
+  const response = await request(fullUrl, {
+    method: "GET"
+  });
+
+  if (response.ok) {
+    const res = await response.json();
+
+    const parsedResult = mPipe(
+      Obj.readKey("collection"),
+      Arr.read,
+      (res) => res[0] as { pageData: string },
+      Obj.readKey("pageData"),
+      (r) => Json.read(r) as { items: string },
+      Obj.readKey("items"),
+      Arr.read,
+      (res) => res[0] as DefaultBlock
+    )(res);
+
+    if (isT(parsedResult)) {
+      return parsedResult;
+    }
+  }
+
+  throw new Error(t("Failed to load kits"));
+};
+
+export const getKitsList = async (url: string): Promise<KitItem[]> => {
+  const response = await request(url, {
+    method: "GET"
+  });
+
+  if (response.ok) {
+    const res = await response.json();
+
+    return res.collections.map((item: { slug: string; title: string }) => ({
+      ...item,
+      id: item.slug
+    }));
+  }
+
+  throw new Error(t("Failed to load kits"));
+};
+
+export const getDefaultLayouts = async (
+  url: string
+): Promise<{
+  templates: LayoutsAPI[];
+  categories: { slug: string; title: string }[];
+}> => {
+  const response = await request(url, {
+    method: "GET"
+  });
+
+  if (response.ok) {
+    const res = await response.json();
+
+    if (res.collections && res.categories) {
+      return { templates: res.collections, categories: res.categories };
+    }
+  }
+
+  throw new Error(t("Failed to load layouts"));
+};
+
+export const getDefaultLayoutsPages = async (
+  url: string,
+  id: string
+): Promise<{
+  collections: LayoutsPageAPI[];
+  paginationInfo: {
+    itemsPerPage: number;
+    lastPage: number;
+    totalCount: number;
+  };
+  styles: Style;
+}> => {
+  const fullUrl = makeUrl(url, {
+    project_id: id,
+    per_page: "20"
+  });
+
+  const response = await request(fullUrl, {
+    method: "GET"
+  });
+
+  if (response.ok) {
+    return response.json();
+  }
+
+  throw new Error(t("Failed to load layouts"));
+};
+
+export const getDefaultLayoutData = async (
+  url: string,
+  layoutId: Literal,
+  id: string
+): Promise<{
+  items: DefaultBlockWithID[];
+}> => {
+  const fullUrl = makeUrl(url, {
+    project_id: layoutId as string,
+    page_slug: id
+  });
+
+  const response = await request(fullUrl, {
+    method: "GET"
+  });
+
+  if (response.ok) {
+    const res = await response.json();
+
+    const parsedResult = mPipe(
+      Arr.read,
+      (res) => res[0] as { pageData: string },
+      Obj.readKey("pageData"),
+      (r) =>
+        Json.read(r) as {
+          items: DefaultBlockWithID[];
+        }
+    )(res);
```

Comment:
create here also an type for `res` to avoi casting, and where you will need to specify type after `Json.read` just create an type predicate and use `pass(yourChecker)` in `mPipe` chain